tools/log_analyser/log_agg.py

```diff
@@ -0,0 +1,49 @@
+import json
+
+
+def seg_log(files, agg_map, start_time, end_time, filename):
```

Comment:
Class: Instead of single long method, A class (eg: LogProcessor) can be defined to encapsulate the functionality.

methods:
Initialization: The __init__ method initializes the object with the provided files, start time, end time, and filename.
Reading Logs: The read_logs method can read and parse the log files, handling JSON decode errors.
Processing Logs: The process_logs method can process the logs based on the timestamp and message content, updating the agg_map accordingly.
Running the Processor: The run method can combine reading and processing of logs, returning the aggregated map.

Reply:
The refactoring of the code is done.